modules/Layout/Layout.tsx

```diff
@@ -48,6 +49,32 @@ function Layout({ children, description, imageUrl, title, hasError }: PropsWithC
         };
     }, []);
 
+    useEffect(() => {
+        function scrollListener() {
+            if (document.body.scrollTop > 300 || document.documentElement.scrollTop > 300) {
+                setIsScrollToTopVisible(true);
+            } else {
+                setIsScrollToTopVisible(false);
+            }
+        }
+        if (typeof window !== 'undefined') {
+            window.onscroll = scrollListener;
+        }
+
+        return () => {
+            if (window !== null) {
+                window.onscroll = null;
+            }
+        };
+    }, []);
+
+    function scrollToTop() {
+        window.scrollTo({
+            top: 0,
+            behavior: 'smooth',
+        });
+    }
+
```

Comment:
IMO all of this logic can be moved into the `ScrollToTopButton` component. It doesn't depend on any state or props in the `Layout` component, so it would be nice to encapsulate everything in one component. Same goes for the `isScrollToTopVisible` state